fast64_internal/oot/scene/exporter/to_c/room_header.py

```diff
@@ -1,78 +1,87 @@
+def getHeaderDefines(outRoom: OOTRoom, headerIndex: int):
     """Returns a string containing defines for actor and object lists lengths"""
-    data = CData()
+    headerDefines = ""
 
-    if len(room.objectIDList) > 0:
-        data.header += f"#define {room.getObjectLengthDefineName(headerIndex)} {len(room.objectIDList)}\n"
+    if len(outRoom.objectIDList) > 0:
+        headerDefines += f"#define {outRoom.getObjectLengthDefineName(headerIndex)} {len(outRoom.objectIDList)}\n"
 
-    if len(room.actorList) > 0:
-        data.header += f"#define {room.getActorLengthDefineName(headerIndex)} {len(room.actorList)}\n"
+    if len(outRoom.actorList) > 0:
+        headerDefines += f"#define {outRoom.getActorLengthDefineName(headerIndex)} {len(outRoom.actorList)}\n"
 
-    return data.header
+    return headerDefines
 
 
 # Object List
+def getObjectList(outRoom: OOTRoom, headerIndex: int):
+    objectList = CData()
+    listName = f"s16 {outRoom.objectListName(headerIndex)}"
 
+    # .h
+    objectList.header = f"extern {listName}[];\n"
 
-def ootObjectListToC(room: OOTRoom, headerIndex: int):
-    data = CData()
-    data.header = "extern s16 " + room.objectListName(headerIndex) + "[];\n"
-    data.source = f"s16 {room.objectListName(headerIndex)}[{room.getObjectLengthDefineName(headerIndex)}]" + " = {\n"
-    for objID in room.objectIDList:
-        data.source += indent + objID + ",\n"
-    data.source += "};\n\n"
-    return data
+    # .c
+    objectList.source = (
+        (f"{listName}[{outRoom.getObjectLengthDefineName(headerIndex)}]" + " = {\n")
+        + ",\n".join(indent + objectID for objectID in outRoom.objectIDList)
+        + ",\n};\n\n"
+    )
 
+    return objectList
 
-# Room Header
-
-
-def ootAlternateRoomMainToC(scene: OOTScene, room: OOTRoom):
-    altHeader = CData()
-    altData = CData()
 
-    altHeader.header = "extern SceneCmd* " + room.alternateHeadersName() + "[];\n"
-    altHeader.source = "SceneCmd* " + room.alternateHeadersName() + "[] = {\n"
-
-    for headerIndex, curHeader in enumerate([room.childNightHeader, room.adultDayHeader, room.adultNightHeader], 1):
+# Room Header
+def getRoomData(outRoom: OOTRoom):
+    roomC = CData()
+
+    roomHeaders = [
+        (outRoom.childNightHeader, "Child Night"),
+        (outRoom.adultDayHeader, "Adult Day"),
+        (outRoom.adultNightHeader, "Adult Night"),
+    ]
+
+    for i, csHeader in enumerate(outRoom.cutsceneHeaders):
+        roomHeaders.append((csHeader, f"Cutscene n°{i + 1}"))
+
+    altHeaderPtrListName = f"SceneCmd* {outRoom.alternateHeadersName()}"
+
+    # .h
+    roomC.header = f"extern {altHeaderPtrListName}[];\n"
+
+    # .c
+    altHeaderPtrList = (
+        f"{altHeaderPtrListName}[]"
+        + " = {\n"
+        + "\n".join(
+            indent + f"{curHeader.roomName()}_header{i:02},"
+            if curHeader is not None
+            else indent + "NULL,"
+            if i < 4
+            else ""
```

Comment:
I'm wondering if this (omitting None cutsceneHeaders) can ever happen, and if it does, does something break?